File: stack/min-stack.java
class MinStack {
    Stack<Integer> st;
    Stack<Integer> mst;
    public MinStack() {
        st=new Stack<>();
        mst=new Stack<>();
    }
    
    public void push(int val) {
        if(st.isEmpty()){
            st.push(val);
            mst.push(val);
        }else{
            st.push(val);
            if(val<=mst.peek()) mst.push(val);
        }
    }
    
    public void pop() {
        int ele=st.pop();
        if(ele==mst.peek()) mst.pop();
    }
    
    public int top() {
        return st.peek();
    }
    
    public int getMin() {
        return mst.peek();
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.push(val);
 * obj.pop();
 * int param_3 = obj.top();
 * int param_4 = obj.getMin();
 */

File: stack/min-stack.py
class MinStack:

    def __init__(self):
        self.st=[]
        self.minst=[]

    def push(self, value: int) -> None:
        self.st.append(value)
        curmin=min(value,self.minst[-1] if self.minst else value)
        self.minst.append(curmin)

    def pop(self) -> None:
        self.st.pop()
        self.minst.pop()

    def top(self) -> int:
        return self.st[-1]

    def getMin(self) -> int:
        return self.minst[-1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(value)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()